### PR TITLE
Hosting features transfer modal: Update data center picker design

### PR DESCRIPTION
**Pull request by allilevine**
Related to https://github.com/Automattic/dotcom-forge/issues/7727

## Proposed Changes

* Update data center picker to match design in issue.

Before | After
---- | ----
Screen Shot 2024-09-18 at 1 59 50 PM | Screen Shot 2024-10-01 at 3 00 27 PM
Screen Shot 2024-09-18 at 2 00 06 PM | Screen Shot 2024-10-06 at 4 14 09 PM

Mobile:

Closed | Open | Selecting
---- | ---- | ----
Screen Shot 2024-10-01 at 3 04 43 PM | Screen Shot 2024-10-06 at 4 13 56 PM | Screen Shot 2024-10-01 at 3 00 40 PM

## Why are these changes being made?

* Improve the data center picker design and keep this interaction exciting for the user.

## Testing Instructions

* Go to /hosting-features for a site that has a Business plan, but isn't yet Atomic.
* Click "Activate now."
* View the form closed state at different screen sizes.
* Click "customize it."
* View the form open state at different screen sizes.
* Pick a data center and click to activate the hosting features. Verify that the site is in the correct data center by going to Sever Settings > Web server settings > Primary data center.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16856818

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @allilevine for including a screenshot in the description! This is really helpful for our translators.

**Comment by allilevine** [comment]
@javierarce This is ready for your review. There are some discrepancies with the design:
- I changed "Your site will be **place** in the optimal data center, but you can customize it." to "Your site will be **placed** in the optimal data center, but you can customize it," to keep the original verb tense.
- Instead of "For redundancy, your site will be replicated in real-time to another region," I kept the original "For redundancy, your site will replicate in real-time to a second data center in a different region." I like the shorter version, but I just feel like we lose too much of the original meaning. 
- I used [the WordPress SelectControl component](https://wordpress.github.io/gutenberg/?path=/docs/components-selectcontrol--docs) for the dropdown. The label and border are different, but aligned with the core component. 

Let me know what you think!

**Comment by allilevine** [comment]
@happychait 👋 This will need a Docs change to https://wordpress.com/support/choose-your-sites-primary-data-center/. Should I open a Docs issue when it ships?

**Comment by happychait** [comment]
@allilevine I logged the docs request (referenced on this PR above). Thanks for the heads up!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16856818).

**Comment by javierarce** [comment]
Thanks, @allilevine!

> I changed "Your site will be place in the optimal data center, but you can customize it." to "Your site will be placed in the optimal data center, but you can customize it," to keep the original verb tense.

Thanks for catching my spelling mistake! 😅 

> Instead of "For redundancy, your site will be replicated in real-time to another region," I kept the original "For redundancy, your site will replicate in real-time to a second data center in a different region." I like the shorter version, but I just feel like we lose too much of the original meaning.

I still think my proposal, "For redundancy, your site will be replicated in real-time to another region," keeps the same meaning as the original text while fitting in just one line. I believe the concept of "replicated to another region" already implies the use of a secondary server in a different location.

> I used [the WordPress SelectControl component](https://wordpress.github.io/gutenberg/?path=/docs/components-selectcontrol--docs) for the dropdown. The label and border are different, but aligned with the core component.

This is a good idea, but for now, I'd still use the same dropdown we are using on other Hosting configuration pages to maintain a consistent look and feel.

---

And after reviewing the changes, I noticed a couple of things:

CleanShot 2024-09-23 at 11 55 30@2x

- On mobile, there’s extra margin under the dropdown.
- The dropdown opens in a weird position (I tested this on Chrome on desktop, so perhaps this issue doesn't occur on an actual mobile device).

**Comment by javierarce** [comment]
The modal looks better in mobile with your last fix, @allilevine, thanks! 

The only thing I'm missing is the copy change I mentioned. Writing "For redundancy, your site will be replicated in real-time to another region. Learn more." would allow us to fit all the text in one line. 
 
CleanShot 2024-10-01 at 16 12 58@2x

File: client/blocks/data-center-picker/index.tsx [hunk 1]
```diff
@@ -23,297 +16,148 @@ interface ExternalProps {
...
-							onChange={ ( event ) => onChange( event.currentTarget.value ) }
-							items={ DataCenterOptions }
-							disabled={ false }
-						/>
-					</div>
+					<StyledSelectControl
+						label={ <StyledLabel>{ translate( 'Pick your primary data center' ) }</StyledLabel> }
+						help={
+							hasTranslation(
+								'For redundancy, your site will be replicated in real-time to another region.'
```
Comment (allilevine) [inline client/blocks/data-center-picker/index.tsx]: @javierarce I added the shorter copy here with a fallback to the translated copy.

**Comment by allilevine** [comment]
> but the border color is challenging to override.

I found a way to override the "backdrop" color in https://github.com/Automattic/wp-calypso/pull/94676/commits/427b9ac36ed9a40b41543bda44ccadfdc62ccd26

File: client/blocks/data-center-picker/README.md [hunk 2]
```diff
@@ -16,4 +16,3 @@ function render() {
 
 - `onChange (func)` - A function to handle the event callback when choosing a data center or picking the recommended one.
 - `value (string)` - The picked data center (this is a controlled component).
-- `compact (boolean)` - Whether or not to display a version of the data center picker without the global map.
```
Comment (allilevine) [inline client/blocks/data-center-picker/README.md:19]: I'm removing the compact version of this component, since we're removing the map.

File: client/blocks/data-center-picker/index.tsx [hunk 3]
```diff
@@ -1,16 +1,9 @@
 import { localizeUrl } from '@automattic/i18n-utils';
 import styled from '@emotion/styled';
-import { Button } from '@wordpress/components';
+import { SelectControl } from '@wordpress/components';
```
Comment (holdercp) [inline client/blocks/data-center-picker/index.tsx:3]: 🎉 I love when standard, systematized components are used.

File: client/blocks/data-center-picker/index.tsx [hunk 4]
```diff
@@ -23,297 +16,148 @@ interface ExternalProps {
...
+const StyledSelectControl = styled( SelectControl )`
+	.components-input-control__backdrop {
+		border-color: var( --studio-gray-10 ) !important;
+	}
+`;
 
-const RECOMMENDED_DATA_CENTER = '';
+const StyledLabel = styled.div`
+	text-transform: none;
+	font-size: 0.875rem;
```
Comment (holdercp) [inline client/blocks/data-center-picker/index.tsx:72]: Do we have vars for things like font sizes too?

File: client/blocks/data-center-picker/index.tsx [hunk 5]
```diff
@@ -23,297 +16,148 @@ interface ExternalProps {
 type Props = ExternalProps & LocalizeProps;
 
 const DataCenterOptions = [
+	{
+		value: '',
+		get label(): string {
+			return translate( 'Optimal data center' );
```
Comment (holdercp) [inline client/blocks/data-center-picker/index.tsx:21]: Just for my learning: if this option is chosen, does the user see which optimal data center was chosen for them in the Server Settings -> Web Server Settings -> Primary Data Center field once the site is atomic?

File: client/blocks/data-center-picker/index.tsx [hunk 6]
```diff
@@ -23,297 +16,148 @@ interface ExternalProps {
...
+	cursor: pointer;
+`;
 
-const CompactPickerSelect = styled( FormSelect )( {
-	width: 'fit-content',
-	borderRadius: '4px',
-} );
+const SupportLink = styled.a`
+	text-decoration: underline;
+	color: var( --color-text-subtle ) !important;
```
Comment (holdercp) [inline client/blocks/data-center-picker/index.tsx]: Not a big fan of using `!important`. Is this style key to the design or UX? If so, that's fine. I just want to ask.

File: client/blocks/data-center-picker/index.tsx [hunk 7]
```diff
@@ -23,297 +16,148 @@ interface ExternalProps {
...
-		boxShadow: '0 0 0 1px var(--color-neutral-light), 0 2px 4px var(--color-neutral-10)',
-	},
-} );
-
-const FormLabelThumbnailImg = styled.img( {
-	objectFit: 'cover',
-} );
+const StyledSelectControl = styled( SelectControl )`
+	.components-input-control__backdrop {
+		border-color: var( --studio-gray-10 ) !important;
```
Comment (holdercp) [inline client/blocks/data-center-picker/index.tsx]: Same Q here about `!important`.

**Review by holdercp — APPROVED** [review]
This looks great! The design is simple and effective, I like all the red in the diff 🟥 .

I have a few questions scattered throughout for either my learning or to just double-check on some things.

I also received this warning in the console when clicking "customize it".
```
console.js:288 Bottom margin styles for wp.components.SelectControl is deprecated since version 6.7 and will be removed in version 7.0. Note: Set the `__nextHasNoMarginBottom` prop to true to start opting into the new styles, which will become the default in a future version.
```
 Looks like certain styling options will be removed from SelectControl in the future. Not sure if it's worth addressing at this point but wanted to make you aware.

File: client/blocks/data-center-picker/index.tsx (hunk 5 above)
Comment (allilevine) [reply client/blocks/data-center-picker/index.tsx:21]: No, looks like we only show it if you've chosen a `geoAffinity` value: 

https://github.com/Automattic/wp-calypso/blob/7c0a42dda287bd6b8474d9d29eedb346a641e258/client/hosting/server-settings/components/web-server-settings-card/index.tsx#L227-L230

We could also try it out and make sure that's the expected behavior. 🙂

File: client/blocks/data-center-picker/index.tsx (hunk 4 above)
Comment (allilevine) [reply client/blocks/data-center-picker/index.tsx:72]: We do, but they're not working here, I'm not sure why. They may not be loaded here:

https://github.com/Automattic/wp-calypso/blob/7c0a42dda287bd6b8474d9d29eedb346a641e258/packages/typography/styles/variables.scss#L38-L48

File: client/blocks/data-center-picker/index.tsx (hunk 6 above)
Comment (allilevine) [reply client/blocks/data-center-picker/index.tsx]: It was to match the design, but I'm not a fan either. I removed those styles.

**Comment by allilevine** [comment]
> This looks great! The design is simple and effective, I like all the red in the diff 🟥 .

@holdercp Thanks for the thorough review! I made some changes. Do you mind re-reviewing?

> I also received this warning in the console when clicking "customize it".
> 
> ```
> console.js:288 Bottom margin styles for wp.components.SelectControl is deprecated since version 6.7 and will be removed in version 7.0. Note: Set the `__nextHasNoMarginBottom` prop to true to start opting into the new styles, which will become the default in a future version.
> ```
> 
> Looks like certain styling options will be removed from SelectControl in the future. Not sure if it's worth addressing at this point but wanted to make you aware.

That's interesting, thanks for pointing it out! I tested locally with these enabled and I don't see any breaking changes, just the `select` getting taller:

```
__nextHasNoMarginBottom
__next40pxDefaultSize
```

There's more about them here: https://wordpress.github.io/gutenberg/?path=/docs/components-selectcontrol--docs

File: client/blocks/data-center-picker/index.tsx [hunk 8]
```diff
@@ -66,13 +65,6 @@ const CustomizeLink = styled.a`
 
 const SupportLink = styled.a`
 	text-decoration: underline;
-	color: var( --color-text-subtle ) !important;
-`;
-
-const StyledSelectControl = styled( SelectControl )`
-	.components-input-control__backdrop {
-		border-color: var( --studio-gray-10 ) !important;
-	}
```
Comment (holdercp) [inline client/blocks/data-center-picker/index.tsx]: 🥇

File: client/blocks/data-center-picker/index.tsx [hunk 9]
```diff
@@ -88,7 +80,14 @@ const DataCenterPicker = ( {
 	value,
 }: Props ) => {
 	const [ isFormShowing, setIsFormShowing ] = useState( false );
-	const { hasTranslation } = useI18n();
+	const hasEnTranslation = useHasEnTranslation();
```
Comment (holdercp) [inline client/blocks/data-center-picker/index.tsx:83]: Why this change?

File: client/blocks/data-center-picker/index.tsx [hunk 10]
```diff
@@ -114,39 +113,25 @@ const DataCenterPicker = ( {
 
 			{ isFormShowing && (
 				<Form>
-					<StyledSelectControl
+					<SelectControl
 						label={ <StyledLabel>{ translate( 'Pick your primary data center' ) }</StyledLabel> }
```
Comment (holdercp) [inline client/blocks/data-center-picker/index.tsx]: I did see [your comment](https://github.com/Automattic/wp-calypso/pull/94676#issuecomment-2395548671) about adding these flags and not seeing any significant visual differences but I don't see where you added the flag in the code. I tested this locally as well and didn't see any differences either but it did silence the warnings.

```suggestion
					<SelectControl
					    __nextHasNoMarginBottom
						label={ <StyledLabel>{ translate( 'Pick your primary data center' ) }</StyledLabel> }
```

**Review by holdercp — APPROVED** [review]
:shipit:

File: client/blocks/data-center-picker/index.tsx (hunk 5 above)
Comment (holdercp) [reply client/blocks/data-center-picker/index.tsx:21]: Confirmed that the chosen data center is not visible in the UI.

File: client/blocks/data-center-picker/index.tsx (hunk 9 above)
Comment (allilevine) [reply client/blocks/data-center-picker/index.tsx:83]: I realized I should use `hasEnTranslation` because it adds a check for EN locale. If EN locale or if the text is translated, show the text. That way it shows up in English, which doesn't need to wait for translations.

File: client/blocks/data-center-picker/index.tsx (hunk 10 above)
Comment (allilevine) [reply client/blocks/data-center-picker/index.tsx]: I tested locally but didn't push the change. It seems weird to me to add these when the warning is functioning as intended, but I can add it in a follow-up to reduce the noise. There's currently another warning on `ToggleControl` that I can fix at the same time:

Screen Shot 2024-10-07 at 1 07 04 PM

It looks like in the past few months we've sometimes added the flag, sometimes not.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16856818).
